src/dvclive/dvc.py

```diff
@@ -46,6 +51,22 @@ def _find_dvc_root(root: Optional[StrPath] = None) -> Optional[str]:
     return None
 
 
+def _find_non_queue_root_dir(root: Optional[StrPath] = None) -> Optional[str]:
+    non_queue_root_dir = _find_dvc_root(root)
+
+    if not non_queue_root_dir:
+        return None
+
+    temp_dir_path = os.path.join(".dvc", "tmp", "exps")
+
+    is_temp_queue_dir = temp_dir_path in non_queue_root_dir
+    if is_temp_queue_dir:
+        non_queue_root_dir = _find_non_queue_root_dir(
+            os.path.abspath(os.path.join(non_queue_root_dir, "..", "..", "..", ".."))
```

Comment:
Happy for pointers on how to make this less ugly.

Reply:
We already do this for the exp name and baseline rev:

https://github.com/iterative/dvc/blob/ef260eaf2e2c207edc26505f510e4f79b11ba6a4/dvc/repo/experiments/queue/base.py#L344-L350

In those lines, you could access `self.repo.root_dir` and it should point to the right place

Reply:
Would that be correct for the temp directories running from the queue? Would it have to be exposed by `dvc queue start`? (I have no idea that's why I'm asking)

Reply:
> Would that be correct for the temp directories running from the queue? Would it have to be exposed by dvc queue start and then would it need to be passed to each worker? (I have no idea that's why I'm asking)

I believe it would be correct in any case. The `run_env` gets serialized when adding experiments to the queue (i.e. `dvc exp run --queue`).
We rely on a mechanism that deserializes and exposes those env vars so the code in the stages can access them.
That is why we can retrieve the right `exp_name` even when there are multiple exps running in the queue

Reply:
This would also mean that users would have to update to the most recent version of DVC and DVCLive when it is released, right?

Reply:
> This would also mean that users would have to update to the most recent version of DVC and DVCLive when it is released, right?

Yes